services/skus/instrumented_datastore.go

```diff
@@ -156,8 +156,18 @@ func (_d DatastoreWithPrometheus) CreateKey(merchant string, name string, encryp
 	return _d.base.CreateKey(merchant, name, encryptedSecretKey, nonce)
 }
 
-func (_d DatastoreWithPrometheus) CreateOrder(ctx context.Context, dbi sqlx.ExtContext, req *model.OrderNew, items []model.OrderItem) (op1 *model.Order, err error) {
-	return _d.base.CreateOrder(ctx, dbi, req, items)
+// CreateOrder implements Datastore
+func (_d DatastoreWithPrometheus) CreateOrder(ctx context.Context, dbi sqlx.ExtContext, oreq *model.OrderNew, items []model.OrderItem) (op1 *model.Order, err error) {
+	_since := time.Now()
+	defer func() {
+		result := "ok"
+		if err != nil {
+			result = "error"
+		}
+
+		datastoreDurationSummaryVec.WithLabelValues(_d.instanceName, "CreateOrder", result).Observe(time.Since(_since).Seconds())
+	}()
+	return _d.base.CreateOrder(ctx, dbi, oreq, items)
```

Comment:
I purposefully did not cover this code with metrics simply because none of them is working. And since it's being overwritten all the time, there was little sense in doing it right. Hence, it was just a stub to satisfy the interface.

If we were to do it properly and useful, the code should have been:

```golang
func (_d DatastoreWithPrometheus) CreateOrder(ctx context.Context, dbi sqlx.ExtContext, oreq *model.OrderNew, items []model.OrderItem) (op1 *model.Order, err error) {
	_since := time.Now()
	defer func() {
		result := "ok"
		if err != nil {
			result = "error"
		}

		datastoreDurationSummaryVec.WithLabelValues(_d.instanceName, "CreateOrder", result).Observe(time.Since(_since).Seconds())
	}()

	op1, err = _d.base.CreateOrder(ctx, dbi, oreq, items)

	return op1, err
}
```

Mind the assignment. Otherwise, the code in the deferred function will always report `ok`, because `err` never gets assigned the result of the call.

Worse than that – it's everywhere in this code base. 😐